clash-prelude/src/Clash/Explicit/Signal.hs

```diff
@@ -859,4 +861,88 @@ sampleWithResetN
 sampleWithResetN nReset nSamples f =
   take nSamples (sampleWithReset nReset f)
 
+-- | Simulate a component until it matches a condition
+--
+-- It prints a message of the form
+--
+-- "Signal sampled for N cycles until value X"
+--
+-- A common usage is with a test bench using
+-- 'Clash.Explicit.Testbench.outputVerifier'. __NB__: Since this uses
+-- 'Clash.Explicit.Testbench.assert', when using @clashi@, read the
+-- note at "Clash.Explicit.Testbench#assert-clashi".
+--
+-- __NB__: This function is not synthesizable
```

Comment:
I think the "A common usage ..." sentence should be moved under the "Example with test bench heading" below and the notes as standalone paragraphs here. In other places where we have notes they each sit by themselves instead of inlined in a paragraph, i.e.

> Some paragraph of text documenting something.
>
> __NB__: This is a note, take note.
> __NB__: This is another note, still important.

Reply:
Ooh, by moving the "A common usage ..." inside that paragraph, that indeed allows me to move the __NB__ to its own paragraph because it is neatly part of the example then. I wasn't happy with it being part of the paragraph, but it couldn't stand on its own either as part of the main text.